src/dom.ts

```diff
@@ -269,3 +268,52 @@ export function compareElementArea(a: HTMLElement, b: HTMLElement): number {
 
     return aArea - bArea
 }
+
+export const hintworthy_js_elems = []
+
+/** Adds or removes an element from the hintworthy_js_elems array of the
+ *  current tab.
+ *  This function is made available to the page's. We should be very careful
+ *  about the input it accepts and what it does with it.
+ */
+export function registerEvListenerAction(elem: EventTarget, add: boolean, event: string) {
+   if (!(elem instanceof Element))
+      return
+   switch (event) {
+      case "click":
+      case "mousedown":
+      case "mouseup":
+      case "mouseover":
+      case "mouseout":
+      case "mouseleave":
+         if (add) {
+               hintworthy_js_elems.push(elem)
+         } else {
+            let index = hintworthy_js_elems.indexOf(elem)
+            if (index >= 0)
+               hintworthy_js_elems.splice(index, 1)
+         }
+   }
+}
+
+/** Replace the page's addEventListener with a closure containing a reference
+ *  to the original addEventListener and our registering function. Do the
+ *  same with removeEventListener.
+ */
+export function hijackPageListenerFunctions(): void {
+   let exportedName = 'registerEvListenerAction'
+   exportFunction(registerEvListenerAction, window, {defineAs: exportedName})
```

Comment:
The newtab and help pages don't get content.js as a content script but as a regular page script. Need to make sure this doesn't break them.

Reply:
Everything seems to work on both the newtab page and the :help page.